### DISCUSSION TITLE
AspNetCore.HealthChecks is heavily used but no longer maintained

### DISCUSSION THREAD

**Discussion by pregress** [microsoft/aspire#9885] [Q&A]
Places where AspNetCore.HealthChecks are used: https://github.com/search?q=repo%3Adotnet%2Faspire+health+language%3AXML+AspNetCore.HealthChecks&type=code

As you can see in this issue: https://github.com/Xabaril/AspNetCore.Diagnostics.HealthChecks/issues/1714 they are looking for maintainers.
Pull requests don't get merged, issues keep stacking.

Is there any plan to help maintain the AspNetCore.HealthChecks or sponsor the current maintainers?

If not will the package be replaced?

**Comment by chuanboz** [comment]
the issue you mentioned for maintainer are not from Microsoft AspnetCore health checks, so you might want to double check. No official message about health check no longer maintained.

**Comment by pregress** [comment]
Although not official asp.net, it's still being used by Aspire everywhere.
Check the dependencies/sbom of aspire: https://github.com/dotnet/aspire/network/dependencies?q=AspNetCore.HealthChecks
This has multiple references to the Xabaril repo, the last release was in february , and I doesn't even support dotnet 9
https://github.com/Xabaril/AspNetCore.Diagnostics.HealthChecks/releases

**Comment by brgrz** [comment]
> the issue you mentioned for maintainer are not from Microsoft AspnetCore health checks, so you might want to double check. No official message about health check no longer maintained.

He meant the UI project which is a community one but referenced in MS docs
https://learn.microsoft.com/en-us/dotnet/architecture/microservices/implement-resilient-applications/monitor-app-health#use-watchdogs

**Comment by pregress** [comment]
No support for .NET 10 in the HealthChecks package. New release and the same problem.
